static/app/utils/useResizable.tsx

```diff
@@ -0,0 +1,170 @@
+import type {RefObject} from 'react';
+import {useCallback, useEffect, useRef, useState} from 'react';
+
+export const RESIZABLE_DEFAULT_WIDTH = 200;
+export const RESIZABLE_MIN_WIDTH = 100;
+export const RESIZABLE_MAX_WIDTH = Infinity;
+
+interface UseResizableOptions {
+  /**
+   * The ref to the element to be resized.
+   */
+  ref: RefObject<HTMLElement | null>;
+
+  /**
+   * The starting size of the container, and the size that is set in the onDoubleClick handler.
+   *
+   * If `sizeStorageKey` is provided and exists in local storage,
+   * then this will be ignored in favor of the size stored in local storage.
+   */
+  initialSize?: number;
```

Comment:
Feels like this shouldn't be something you set, the CSS should just imply this no?